cpp/src/utilities/column_utils.hpp

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright 2019 BlazingDB, Inc.
+ *     Copyright 2019 Eyal Rozenberg <eyalroz@blazingdb.com>
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * @file Small utility functions for working with @ref gdf_column structures,
+ * which are raw C structures without any guarantees and operators defined for
+ * them, and thus require a bit of care when handling. This care and due
+ * diligence tends to be rather repetitive; hence this file.
+ *
+ * @note since the "right" way to work with these structures is pass references,
+ * but the "C" way of doing the same involves pointers, many functions are
+ * implemented for both.
+ */
+#ifndef UTILITIES_COLUMN_UTILS_HPP_
+#define UTILITIES_COLUMN_UTILS_HPP_
+
+#include <utilities/type_dispatcher.hpp>
+
+#include <cudf.h>
+
+#include <algorithm>
+
+namespace cudf {
+
+template<typename T>
+T* get_data(const gdf_column& column) noexcept { return static_cast<T*>(column.data); }
+
+template<typename T>
+T* get_data(const gdf_column* column) noexcept { return get_data<T>(*column); }
+
+namespace detail {
+
+struct integral_check {
+   template <typename T>
+   constexpr bool operator()(void) const noexcept
+   {
+      return std::is_integral<T>::value;
+   }
+};
+
+} // namespace detail
+
+constexpr inline bool is_an_integer(gdf_dtype element_type) noexcept
+{
+   return cudf::type_dispatcher(element_type, detail::integral_check{});
+}
+
+constexpr inline bool is_integral(const gdf_column& column) noexcept { return is_an_integer(column.dtype); }
+constexpr inline bool is_integral(const gdf_column* column) noexcept{ return is_integral(*column); }
+
+constexpr bool is_nullable(const gdf_column& column) noexcept { return column.valid != nullptr; }
```

Comment:
Could you also add a utility function like `has_nulls` that first checks if the column `is_nullable` and then returns if `null_count > 0`?

Reply:
I don't know about that. I mean, if the column is not known to be valid, just `validate()` it when you get it. If it is, `null_count > 0` works. But - you can convince me this would be a useful addition.

Reply:
> If it is, null_count > 0 works.

True, but `col->valid != nullptr` also just works in place of `is_nullable(col)`. 

The point is to abstract away the implementation details so that you don't have to think about them and provide functions that contribute to a nicer overall narrative flow. 

As far as naming, I don't think it's ambiguous.

```
is_nullable(col) // this means a column _can_ contain nulls
has_nulls(col) // this means a column _does_ contain nulls
```

Reply:
@jrhemstad : Well, if you think people will use it, then ok.

Reply:
Just `const&` is fine with me.